x-pack/plugins/security_solution/public/timelines/components/side_panel/event_details/index.tsx

```diff
@@ -51,10 +53,18 @@ const EventDetailsPanelComponent: React.FC<EventDetailsPanelProps> = ({
   timelineId,
   isReadOnly,
 }) => {
+  // The referenced alert _index is .internal.alerts-security.alerts-spaceId,
+  // but we always want to use the alias .alerts-security.alerts-spaceId.
+
+  const currentSpaceId = useSpaceId();
+  const eventIndex = expandedEvent.indexName.includes(DEFAULT_ALERTS_INDEX)
```

Comment:
I'm not sure, but curious if there are thoughts on making this an explicit swap for `.internal.alerts rather than an includes`?

Reply:
Consider adding the following _failing_ test to `x-pack/plugins/security_solution/public/timelines/components/side_panel/event_details/index.test.tsx`

```ts
    test(`it does NOT use the alerts alias when regular events happen to include a trailing '${DEFAULT_ALERTS_INDEX}' in the index name`, () => {
      const indexName = `.ds-logs-endpoint.alerts-default-2022.08.09-000001${DEFAULT_ALERTS_INDEX}`; // a regular event, that happens to include a trailing `.alerts-security.alerts`
      const propsWithEventIndex = {
        ...defaultProps,
        expandedEvent: {
          eventId: ecsData._id,
          indexName,
        },
      };

      render(
        <TestProviders>
          <EventDetailsPanel {...{ ...propsWithEventIndex }} />
        </TestProviders>
      );

      expect(useTimelineEventsDetails).toHaveBeenCalledWith({
        entityType: 'events',
        indexName, // <-- use the original index name, not the alerts alias
        eventId: propsWithEventIndex.expandedEvent.eventId,
        runtimeMappings: mockRuntimeMappings,
        skip: false,
      });
    });
```

The above test fails with the current `includes` because:

- expected: `.ds-logs-endpoint.alerts-default-2022.08.09-000001.alerts-security.alerts`
- actual: `alerts-security.alerts-testSpace`

The failing test above passes if the following suggested change is implemented:

```suggestion
  const eventIndex = expandedEvent.indexName.startsWith(`.internal${DEFAULT_ALERTS_INDEX}`)
```

Consider adding the failing test and then replacing `includes` with `startsWith` per the suggestion above.

Reply:
Yea, makes sense. I'll also need to add `.preview` for the rule preview flyout as well